.github/workflows/run-qit.yml

```diff
@@ -0,0 +1,66 @@
+name: Run QIT
+
+# **What it does**: Runs a suite of QIT tests for the extension.
+# **Why we have it**: To be able to check QIT compatibility at once. For example, to test a specific branch, or upcoming release.
+
+on:
+  workflow_dispatch:
+    inputs:
+      # Configure which tests to run.
+      test-activation:
+        description: 'Should activation be tested?'
+        required: true
+        default: true
+        type: boolean
+      test-security:
+        description: 'Should security be tested?'
+        required: true
+        default: true
+        type: boolean
+      test-phpstan:
+        description: 'Should phpstan be tested?'
+        required: true
+        default: true
+        type: boolean
+      test-api:
+        description: 'Should API be tested?'
+        required: true
+        default: true
+        type: boolean
+      test-e2e:
+        description: 'Should E2E be tested? (takes a lot of time)'
+        required: true
+        default: false
+        type: boolean
+
+      # Advanced customization.
+      options:
+        description: 'Additional options for `qit` command, like `--optional_features=hpos`.'
+        required: false
+
+  # Used to test this PR, to be removed after before merging.
+  push:
+    branches:
+      - dev/qit-workflow
+
+jobs:
+  build:
+    name: Build extension
+    uses: ./.github/workflows/build.yml
+    secrets: inherit
```

Comment:
If we generated the artifact? Why do we need to run again the build? Shouldn't we download the artefact instead?

Reply:
> If we generated the artifact

This is the only job in this workflow that generates an artifact.

-----

I do see we have a redundancy in general. As for example, for release PR, we run the build multiple times:
- once in general, build for bundle size on  PR trigger - https://github.com/woocommerce/google-listings-and-ads/blob/trunk/.github/workflows/build.yml#L3-L9
- once running local E2E on release PR trigger - https://github.com/woocommerce/google-listings-and-ads/blob/trunk/.github/workflows/e2e-tests.yml#L4-L46
- once here, we would also call it for release PRs.

However, to solve that, we would need to restructure our existing workflows, which is out of the scope of this PR.

We would need to make workflow interact with each other more, fetching their artifacts, controlling concurrency, etc. But I think we could benefit from taking one step back, and doing a more overall change.

:bulb: 
Suppose we redesign our workflows to distinguish reusable, composable blocks ("what") from actual flow definitions ("when _ then _"). Something like:


> reusable blocks: // What **work** should be done (technically **work**flows, jobs, & actions)
> 🏗 build package
> 👁 apply bundle watch
> 🧪 run local E2E tests
> 🆏 run QIT
> 🚀 release `gha-dev-build`
> 📖 deploy storybook
> 
> 
> work**flows**: // When what should happen - the actual flow
> - On Release PR: 🏗 , (👁, 🧪, 🆏)
> - On regular PR 🏗 , 👁
> - On push to develop 🏗 , (👁, 🧪, 🆏, 📖, 🚀)

- 
It would be easier to avoid redundancy, parallelize, and IMHO reason & maintain.
As we start to reason from the flow perspective, 
Q: "What is the process that we automate?",  
A: "release" / "regular feature development" / "develop branch consolidation" / "compatibility checks"
Q: "When do we do that?"
A: "for `release/` branch" / "for any PR" / "for pushes to `develop`" / "manual call"
Q: "What do we do?"
A: "This then that"

Instead of what we have currently:
```
run build, bundle watch, and conditional `gha-dev-build` publish on pushes and all pull requests
run storybook on pushes to develop & trunk
run build & E2E on pushes and PRs to `release/*` & manual calls
run build & QIT on manual calls, pushes to develop, and release PRs
```
When we reason only from the perspective of job, that's done. 
Q: "What do we do?"
A: "Build the package to watch its size & optionally publish it" / "Build the package to run E2E tests" / ..
Q: "When do we do that?"
A: "On releases, but also on regular pushes, but then we do only part of it. We also do the thing when we do that other thing..." :confused: 
Q: "What is the process that we automate?"
A: ":thinking: ... work?"

But that's a discussion and project on its own :)

Reply:
Hey @tomalec sorry maybe I expressed wrong my question.

I see this PR generates an artifact in the build workflow.

Why is this for? 

My assumption was that it was needed for the new Qit workflow... but I see we are running the full build again instead of downloading the artifact.

Reply:
`woocommerce/grow/.github/workflows/run-qit-extension.yml@add/qit-workflows` 
does not run the build, it downloads the artifact :confused: 

https://github.com/woocommerce/grow/pull/81/files#diff-f653f2a856040f6c20d52799af46664b5dc76e12be91f8443c4f04f22bdf052dR78-R82

The workflow here `.github/workflows/run-qit.yml`:

0. On manual trigger
1. Runs the build workflow (`.github/workflows/build.yml`)
2. Runs the QIT which uses the artifact (`woocommerce/grow/.github/workflows/run-qit-extension.yml`)

Then we also have a case when the build is run on other triggers `.github/workflows/build.yml`:

0. On PRs and pushes to trunk/develop
1. Run the build workflow